File: container_with_most_water/container_with_most_water.go
package container_with_most_water

func maxArea(height []int) int {
	length := len(height)
	left := 0
	right := length - 1
	mArea := 0
	for left < right {
		min := height[left]
		if min > height[right] {
			min = height[right]
			right--
		} else {
			left++
		}
		area := (right - left + 1) * min
		if area > mArea {
			mArea = area
		}

	}
	return mArea
}

File: container_with_most_water/container_with_most_water.ts
function maxArea(height: number[]): number {
    let len = height.length;
    let maxArea = 0;
    let left = 0, right = len - 1;

    while (left < right) {
        let shortBorder = height[left];
        if (shortBorder > height[right]) {
            shortBorder = height[right]
            right--
        } else {
            left++
        }
        const area = (right - left + 1) * shortBorder
        if (maxArea < area) {
            maxArea = area
        }
    }
    return maxArea

};
